### DISCUSSION TITLE
Looks like `__` helper has same condition with proxied `trans`

### DISCUSSION THREAD

**Discussion by oleghalin** [laravel/framework#47020] [Q&A] [answered]
Maybe it makes sense just proxy `__()` to `trans()`? Why do we need additional condition here? 

https://github.com/laravel/framework/blob/d1f7c9e802930a6ccc0567061252a2bedcbacf61/src/Illuminate/Foundation/helpers.php#L953

**Comment by henzeb** [comment, marked as answer]
Because trans returns a translator instance when $key is null, and __ just returns the key.

**Reply by oleghalin** [reply]
What idea behind translator implementation returned from helper? Honestly i don't see any use case when such code makes sense and easy to read. Atleast as soon as `trans()` accepts arguments..
`trans()->get("some.jey")`

**Reply by henzeb** [reply]
Honestly, you can do more than just getting translations. You can also set the locale, for example.

It is however impossible to do this for the current laravel version, as it would be a breaking change.

**Reply by oleghalin** [reply]
It possible without any backward issue. Just run app('translator')->get($key..) instead of proxying to `trans()`
